### PR TITLE
[Dy2stat] support usage: to_static(model)

### PR DESCRIPTION
**Pull request by Aurelius84**
### PR types

New features
### PR changes

APIs
### Describe

support usage: to_static(model)

### Code Example:
Before this PR，if we want decorate a `forward()` from model, we shall write code like:
```python
model.forward = to_static(model.forward, input_spec=...)
``` 
It's not easy-to-use. In this PR，we support decorate model directly and convert call for `foward` and all inner function.

```Python
class SimpleNet(Layer):
    def __init__(self, ...):
        .......

    def forward(self, x):
        ......
        return out

net = SimpleNet()
x = paddle.to_tensor(np.ones(4, 10))

# case 1:  decorate model and feed data
net = to_static(net)
out = net(x)
print(out)

# case 2: decorate model with InputSpec
net = to_static(net, input_spec=[InputSpec([None, 8, 10])])
self.assertListEqual(list(input_shape), [-1, 8, 10])


# case 3: redecorate with new InputSpec
net = declarative(net, input_spec=[InputSpec([None, 16, 10])])
input_shape = net.forward.inputs[0].shape
self.assertListEqual(list(input_shape), [-1, 16, 10])
```

### CODE REVIEW

File: python/paddle/fluid/dygraph/jit.py [hunk 1]
```diff
@@ -210,7 +210,16 @@ def decorated(python_func):
 
     # for usage: `declarative(foo, ...)`
     if function is not None:
-        return decorated(function)
+        if isinstance(function, Layer):
+            if isinstance(function.forward, StaticLayer):
```
Comment (zhhsplendid) [inline python/paddle/fluid/dygraph/jit.py:214]: Please check whether ``function`` has ``forward``, if not, I think we should throw our exception to indicate users?

Comment (zhhsplendid) [reply python/paddle/fluid/dygraph/jit.py:214]: Sorry, I was wrong, ``function`` is an instance of ``Layer`` so it must have the ``forward``. I will approve this PR

**Review by zhhsplendid — APPROVED** [review]
LGTM

File: python/paddle/fluid/dygraph/jit.py (hunk 1 above)
Comment (Aurelius84) [reply python/paddle/fluid/dygraph/jit.py:214]: It inherit `Layer` and hold `forward` function.

File: python/paddle/fluid/dygraph/jit.py [hunk 2]
```diff
@@ -210,7 +210,16 @@ def decorated(python_func):
...
+                class_name = function.__class__.__name__
+                warnings.warn(
+                    "`{}.forward` has already been decorated somewhere. It will be redecorated to replace previous one.".
+                    format(class_name))
```
Comment (liym27) [inline python/paddle/fluid/dygraph/jit.py:218]: > 这次PR中不重要

动转静中的warning，统一了接口，后续建议使用 `logging_utils._TRANSLATOR_LOGGER.warn()`

我这边最近会统一做更改

**Review by liym27 — APPROVED** [review]
LGTM

File: python/paddle/fluid/dygraph/jit.py (hunk 2 above)
Comment (Aurelius84) [reply python/paddle/fluid/dygraph/jit.py:218]: 赞

**Pull request merged**